### PR TITLE
Update MNIST to use the deep learning library.

### PR DESCRIPTION
**Pull request by rxwei**

### CODE REVIEW

File: MNIST/MNIST.swift [hunk 1]
```diff
@@ -12,115 +12,86 @@
...
+            let loss = softmaxCrossEntropy(logits: ŷ, labels: y)
+            totalLoss += loss.scalarized()
+            return loss
         }
-        print("""
-              [Epoch \(epoch)] Accuracy: \(correctGuesses)/\(totalGuesses) \
-              (\(Float(correctGuesses) / Float(totalGuesses)))
-              """)
+        // Update the model's differentiable variables along the gradient vector.
+        optimizer.update(&classifier.allDifferentiableVariables, along: 𝛁model)
```
Comment (dan-zheng) [inline MNIST/MNIST.swift]: Should this be negative?
```suggestion
        optimizer.update(&classifier.allDifferentiableVariables, along: -𝛁model)
```

Comment (rxwei) [reply MNIST/MNIST.swift]: No, because optimizers are responsible for doing minimization given gradients. This is also why `RiemannSGD.update(_:along:)` is doing `.zero - tangent`, speaking of which, the `along:` argument label is not super appropriate.

Comment (dan-zheng) [reply MNIST/MNIST.swift]: Aha, I see. I agree that `along:` is not quite accurate.
